src/main/java/com/mycompany/myapp/config/SecurityConfiguration.java

```diff
@@ -63,19 +74,47 @@ public void configure(HttpSecurity http) throws Exception {
         .and()
             .authorizeRequests()
             .antMatchers("/api/**").authenticated()
+            .antMatchers("/authorize").authenticated()
             .antMatchers("/management/health").permitAll()
             .antMatchers("/management/info").permitAll()
-            .antMatchers("/management/**").hasAuthority(AuthoritiesConstants.ADMIN);
-
+            .antMatchers("/management/**").hasAuthority(AuthoritiesConstants.ADMIN)
+        .and()
+            .oauth2Login();
     }
 
-    /**
-     * This OAuth2RestTemplate is only used by AuthorizationHeaderUtil that is currently used by TokenRelayRequestInterceptor
-     */
+
     @Bean
-    public OAuth2RestTemplate oAuth2RestTemplate(OAuth2ProtectedResourceDetails oAuth2ProtectedResourceDetails,
-        OAuth2ClientContext oAuth2ClientContext) {
-        return new OAuth2RestTemplate(oAuth2ProtectedResourceDetails, oAuth2ClientContext);
+    @SuppressWarnings("unchecked")
+    public GrantedAuthoritiesMapper userAuthoritiesMapper() {
```

Comment:
Hey @mraible,

Thanks for giving this example, I love looking at your work as a point of reference.

Question for you around converting the groups claim to the authorities in the security context. Any reason for not just using a custom mapper in keycloak to map the groups/client-roles/realm-roles to a claim labelled as one of the `WELL_KNOWN_SCOPE_ATTRIBUTE_NAMES`[1] and letting spring do its conversion?

[1] https://github.com/spring-projects/spring-security/blob/master/oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/authentication/JwtGrantedAuthoritiesConverter.java#L40